File: java/pattern/src/proxy/dynamicproxy/RealStar.java
package proxy.dynamicproxy;

public class RealStar implements Star {
    @Override
    public void confer() {
        System.out.println("真实角色，面谈");
    }

    @Override
    public void signContract() {
        System.out.println("真实角色，签合同");
    }

    @Override
    public void bookTicket() {
        System.out.println("真实角色，订票");
    }

    @Override
    public void sing() {
        System.out.println("真实角色，唱歌");
    }

    @Override
    public void collectMoney() {
        System.out.println("真实角色，收钱");
    }
}

File: php/proxy/RealStar.php
<?php

/**
 * 真实明星
 * 
 */
class RealStar implements Star
{
    // 面谈
    public function confer()
    {
        echo 'real.confer', PHP_EOL;
    }
    // 签合同
    public function signContract()
    {
        echo 'real.signContract', PHP_EOL;
    }
    // 订票
    public function bookTicket()
    {
        echo 'real.bookTicket', PHP_EOL;
    }
    // 唱歌
    public function sing()
    {
        echo '真实明星，唱歌！', PHP_EOL;
    }
    // 收钱
    public function collectMoney()
    {
        echo 'proxy.collectMoney', PHP_EOL;
    }
}
